### PR TITLE
Switch let/const to var in the scanner & parser for top-levelish variables.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
This pull request regains some of the performance we lost in #51387 when we switched our emit from ES5 to ES2018 - a significant part due to TDZ checks that engines need to perform. These TDZ checks can often be eliminated, but engines probably have a hard time knowing whether a function can eliminate these checks.

By switching many of the top-level shared closure variables in the parser and scanner from `let`/`const` to `var`, we're able to eliminate those checks. Doing so should be fairly safe, since our functions follow the same initialization mechanism of declaring variables, and then calling the "workhorse" function afterwards.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot perf test this faster

**Comment by DanielRosenwasser** [comment]
@typescript-bot perf test this faster

**Comment by jakebailey** [comment]
Yay!!!

**Comment by jakebailey** [comment]
Aside: is it _just_ the scanner? Can we revert parser and test only the scanner? Is it a specific variable? That would be very interesting to know.

**Review by jakebailey — APPROVED** [review]
:shipit:

File: src/compiler/scanner.ts [hunk 1]
```diff
@@ -1031,6 +1032,8 @@ export function createScanner(languageVersion: ScriptTarget,
         scanRange,
     };
 
+    /* eslint-disable no-var */
```
Comment (bakkot) [inline src/compiler/scanner.ts:1035]: Drive-by - was this maybe supposed to be `eslint-enable`?

Comment (jakebailey) [reply src/compiler/scanner.ts:1035]: Uh oh, sure was!
